remove-other-tokens.py

```diff
@@ -0,0 +1,90 @@
+#!/opt/privacyidea/bin/python
+
+from privacyidea.lib.token import get_tokens, get_one_token, remove_token
+from privacyidea.lib.user import User
+import argparse
+from privacyidea.app import create_app
+import logging
+
+__doc__ = """
+This scripts create new tokens for a given user.
+#
+This is a script that can be called by the privacyIDEA script handler. It
+is designed to run as a post event handler at /token/init. It will remove
+all tokens of the user but the token, which was just enrolled.
+
+The script can be configured to remove only tokens which share the type
+with the token that was just enrolled. It can act on all tokens or on the
+active ones only.
+
+It takes the arguments
+
+   remove-other-tokens.py --user <user> --realm <realm> --serial <token serial>
+
+You can place the script in your scripts directory /etc/privacyidea/scripts/
+and use it in the script event handler. It logs with level debug to the
+privacyidea log file.
+
+Adapt it (like the REMOVE_OTHER_TOKENS_PER and ONLY_ACTIVE) to your needs.
+
+(c) 2021, Henning Hollermann <henning.hollermann@netknights.it>
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License version 3 as
+    published by the Free Software Foundation.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program; if not, write to the Free Software
+    Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+"""
+
+# "type": remove other tokens that share the type of the token in the request
```

Comment:
Would *all* tokens of this type be removed? Or only the tokens of this very user with the given type?
Please adapt the description a bit.

Reply:
only tokens of the given user, except the one given by serial will be removed.